pkg/driver/node/mounter/systemd_mounter.go

```diff
@@ -123,30 +131,19 @@ func (m *SystemdMounter) Mount(bucketName string, target string, credentials *Mo
 		return fmt.Errorf("Could not check if %q is a mount point: %v, %v", target, statErr, err)
 	}
 
-	if isMountPoint {
-		klog.V(4).Infof("NodePublishVolume: Target path %q is already mounted", target)
-		return nil
-	}
-
 	env := envprovider.Default()
-	var authenticationSource AuthenticationSource
-	if credentials != nil {
-		var awsProfile awsprofile.AWSProfile
-		if credentials.AccessKeyID != "" && credentials.SecretAccessKey != "" {
-			// Kubernetes creates target path in the form of "/var/lib/kubelet/pods/<pod-uuid>/volumes/kubernetes.io~csi/<volume-id>/mount".
-			// So the directory of the target path is unique for this mount, and we can use it to write credentials and config files.
-			// These files will be cleaned up in `Unmount`.
-			basepath := filepath.Dir(target)
-			awsProfile, err = awsprofile.CreateAWSProfile(basepath, credentials.AccessKeyID, credentials.SecretAccessKey, credentials.SessionToken)
-			if err != nil {
-				klog.V(4).Infof("Mount: Failed to create AWS Profile in %s: %v", basepath, err)
-				return fmt.Errorf("Mount: Failed to create AWS Profile in %s: %v", basepath, err)
-			}
-		}
```

Comment:
This logic to create AWS profile based on long-term credentials has been moved to `provideLongTermCredentialsFromDriver` function in `pkg/driver/node/credentialprovider/provider_driver.go`.